### ISSUE TITLE
Tooltips for secondary objective quests are not updated/removed upon completion.

### ISSUE THREAD

**Issue by Cabro** [Questie/Questie#3466]
# Bug description

Tooltip for secondary objective will still show even though you completed the quest.
The map/minimap icons are correctly removed.

# Screenshots
![imgur](https://i.imgur.com/XcXQU3F.png)

# Questie & WoW version
6.10.0 TBC

**Comment by Laumesis** [comment]
So tooltip stuff of both `extraObjectives` and `requiredSourceItems` are not removed once `isComplete ~= nil` (`1`or `-1`). Correct?

I think it will be easy fix after few other fixes / refactors are in place.

**Comment by Laumesis** [comment]
Update:
This is in line with all other tooltip stuff. A quest and its objectives are kept in tooltips until it gets completed (<- blizzard word) (read: turned in).
The tooltip text should get changed to green once quest is complete. That's a bug at least.

Some with more power may comment on if intended behavior should be changed.

@Gogo1951 please do not reply, this is not to you.

**Comment by BreakBB** [comment]
As @Laumesis already said, the current behaviour is streamlined to the regular objectives.

I agree that at least the secondary objectives are not useful to show once a quest is completed (not turned it yet).

Even though this is not a bug as it is the intended behaviour, we should change it 👍🏻

**Comment by Dyaxler** [comment]
Opps... forgot to reference this along with #4948 

I'm almost 100% certain this is the same bug... however, since MY fix in #4948 pertains to Quest Finishers, these tooltips will still be present.

**Comment by Cabro** [comment]
I guess just wait and see what BreakBB says

**Comment by Dyaxler** [comment]
Yeah... because personally I think we should leave them. For "Normal" quests we leave the Objectives intact until AFTER the player turns in the Quest. Which is not broken (removing tooltips from Quest Finishers and Objectives after the quest is turned in).

**Comment by Dyaxler** [comment]
@BreakBB Do you agree with my comment? We should leave these tooltips alone since they are attached to an Objective "creature" and they aren't duplicated.

**Comment by MyTechnoHunter** [comment]
> @BreakBB said :
> As @Laumesis already said, the current behaviour is streamlined to the regular objectives.
> 
> I agree that at least the secondary objectives are not useful to show once a quest is completed (not turned it yet).
> 
> Even though this is not a bug as it is the intended behaviour, we should change it 👍🏻


sounds like they are for hiding the things.. BUT i dont know that we emphasized the quest cases where the extraObjective tells how to finish the quest i think it was? it was in the convo somewhere, i think in discord..

**Comment by Dyaxler** [comment]
I'm leveling a toon in Howling Fjord... there should be plenty of Quests with extraObjectives I can test with. I COULD modify the `QuestieQuest:AddFinisher()` code that I used to find and nuke duplicate tooltips to find and strip out any and all related extraObjective tooltips based on QuestID... it would be easy enough if everyone agrees.

**Comment by BreakBB** [comment]
When #4948 is fixed, that is already a good step forward.

Also: We do have quests which are complete on accept, but we do want to show the extraObjectives. There are multiple quests for example (TBC and Wotlk) which let you get faster to the turn in NPC when talking to someone and fly over to that NPC.

So I **now** don’t think we should change this behaviour.

BUT there is an integer field in the extraObjectives table already which we want to use to hide an extraObjective when an objective is complete (e.g. objective 1 is complete so we put a 1 in that field). We simply never added the logic to handle that field.

**Comment by Cabro** [comment]
That is a really great way to manage which extraobjectives to show and not! Once the code is edited to handle the logic, i would go and fix them in corrections, they're not that many extraobjectives in the files

**Comment by BreakBB** [comment]
The PR linked above only hides the map icons. In a second step the tooltips also need to be removed for complete extraObjectives.
